src/components/scene/device-motion-permission-ui.js

```diff
@@ -0,0 +1,159 @@
+    this.onOrientationChangeClick = bind(this.onOrientationChangeClick, this);
+    this.grantedDeviceMotion = bind(this.grantedDeviceMotion, this);
+    if (typeof window.orientation !== 'undefined') {
+      try {
+        /*eslint-disable */
+        if (
+          DeviceOrientationEvent &&
+          typeof DeviceOrientationEvent.requestPermission === 'function'
+        ) {
+          DeviceOrientationEvent.requestPermission()
+            .then(response => {
+              if (response === 'granted') {
+                this.grantedDeviceMotion();
+              }
+            })
+            /* eslint-enable */
+            .catch(err => {
+              console.log(err);
+              this.deviceMotionEl = createDeviceMotionPermissionWindow(
+                this.onDeviceMotionClick,
+                this
+              );
+              this.el.appendChild(this.deviceMotionEl);
+            });
+        } else {
+          this.grantedDeviceMotion();
+        }
+      } catch (oops) {
+        this.grantedDeviceMotion();
+      }
+    } else {
+      this.remove();
+    }
+  },
+
+  remove: function () {
+    if (this.deviceMotionEl) {
+      this.el.removeChild(this.deviceMotionEl);
+    }
+  },
+
+  /**
+   * Enable device motion permission when clicked.
+   */
+  onDeviceMotionClick: function () {
+    try {
+      /*eslint-disable */
+      if (
+        DeviceOrientationEvent &&
+        typeof DeviceOrientationEvent.requestPermission === 'function'
+      ) {
+        DeviceOrientationEvent.requestPermission()
+          .then(response => {
+            if (response === 'granted') {
+              this.grantedDeviceMotion();
+            } else {
+              console.log('Device Motion permission not granted.');
+            }
+          })
+          .catch(console.error);
+          /* eslint-enable */
+      } else {
+        this.grantedDeviceMotion();
+      }
+    } catch (oops) {
+      console.log(
+        'Your device and application combination do not support device motion events.'
+      );
+    }
+  },
+
+  grantedDeviceMotion: function () {
+    const orientationChange=this.el.getAttribute('orientation-change');
+    const deviceOrientation=this.el.getAttribute('device-orientation');
+    this.remove();
+      if(orientationChange){
+        window.addEventListener('orientationchange', e => {
+          window[orientationChange](e);
+        });
+      }
+      if(deviceOrientation){
+        window.addEventListener('deviceorientation', e => {
+          window[deviceOrientation](e);
+        });
+      }
+    }
+  }
+);
+
+/**
+ * Create a button that when clicked will provide device motion permission.
+ *
+ * Structure: <div><button></div>
+ *
+ * @param {function} onClick - click event handler
+ * @returns {Element} Wrapper <div>.
+ */
+
+function createDeviceMotionPermissionWindow (onClick, obj) {
+  var wrapper, innerWrapper, aframeBuilt;
+  var cancelBtn, continueBtn;
```

Comment:
We do one variable declaration per line and we avoid abbreviations: cancelButton and cotinueButton